Python/AthleteReportForce.py

```diff
@@ -0,0 +1,264 @@
+LTurns = findLeftTurns(dat.RTotal_Filt, dat.LTotal_Filt)
+
+RTurns = cleanTurns(RTurns)
+LTurns = cleanTurns(LTurns)
+
+makeTurnPlot(dat, LTurns, 'Left Turns')
+makeTurnPlot(dat, RTurns, 'Reft Turns')
+answer = messagebox.askyesno("Question","Is data clean?")
+
+if answer == False:
+    plt.close('all')
+    print('Adding file to bad file list')
+
+if answer == True:
+    plt.close('all')
+    print('Estimating point estimates')
+
+
+for i, value in enumerate(LTurns):
+# Loop through all cleaned turns to calculate discrete outcome measures.
+# using right side only (left turns) first
+
+    if i < len(LTurns) - 2:
+        # index of peak turn. saved for later in case we want to use it
+        pkIdx = np.argmax(dat.RTotal_Filt[LTurns[i]:LTurns[i+1]])
+        pkIdx = value + pkIdx
+        ## Extract relevent parameters from a turn here ##
+        peakRForce.append(np.max(dat.RTotal_Filt[LTurns[i]:LTurns[i+1]]))
+    
+
+    
+for i, value in enumerate(RTurns):
+# Loop through all cleaned turns to calculate discrete outcome measures.
+# using right side only (left turns) first
+
+    if i < len(RTurns) - 2:
+    # index of peak turn. saved for later in case we want to use it
+        pkIdx = np.argmax(dat.LTotal_Filt[RTurns[i]:RTurns[i+1]])
+        pkIdx = value + pkIdx
+        ## Extract relevent parameters from a turn here ##
+        peakLForce.append(np.max(dat.LTotal_Filt[RTurns[i]:RTurns[i+1]]))
+
+
+
+SMALL_SIZE = 14
+MEDIUM_SIZE = 16
+BIGGER_SIZE = 18
+
+plt.rc('font', size=SMALL_SIZE)          # controls default text sizes
+plt.rc('axes', titlesize=SMALL_SIZE)     # fontsize of the axes title
+plt.rc('axes', labelsize=MEDIUM_SIZE)    # fontsize of the x and y labels
+plt.rc('xtick', labelsize=SMALL_SIZE)    # fontsize of the tick labels
+plt.rc('ytick', labelsize=SMALL_SIZE)    # fontsize of the tick labels
+plt.rc('legend', fontsize=SMALL_SIZE)    # legend fontsize
+plt.rc('figure', titlesize=BIGGER_SIZE)  # fontsize of the figure title
+bar_colors = ['tab:blue','tab:red']
+
+
+rightAvgForce = np.mean(peakRForce)
+leftAvgForce = np.mean(peakLForce)
+
+pctDiff = abs(100 * ((rightAvgForce - leftAvgForce) / ( (rightAvgForce+leftAvgForce) / 2)))
+if rightAvgForce > leftAvgForce:
+    print("Skier has ", round(pctDiff), " % greater force on right foot (left turns with right foot downhill are stronger")
+
+if leftAvgForce > rightAvgForce:
+    print("Skier has ", round(pctDiff), "% greater force on left foot (right turns with left foot downhill are stronger")
+
+
+fig, (ax) = plt.subplots(1,1)
+variab = ('Left', 'Right')
+performance = [leftAvgForce, rightAvgForce]
+error = [np.std(peakLForce), np.std(peakRForce)]
+ax.bar(variab, performance,yerr=error, color = bar_colors)
+ax.set_xlabel('Side')
+ax.set_title('Peak Force During Turn')
+ax.set_ylabel('Force (N)')
+variab = ('Left', 'Right')
+plt.tight_layout()
+plt.show()
+
+# Save the trial segmentation
+trial_segment = np.array([trimName,pts])
```

Comment:
may want to have an if statement here regarding if the particular trial already has an associated numpy file. OR on line 151 include "pts = trial_segment_old[1]" that ways the numpy variable can be saved over again.